src/cluster.c

```diff
@@ -852,176 +689,312 @@ static dict *parse_cluster_slots(valkeyClusterContext *cc, valkeyReply *reply) {
     return NULL;
 }
 
-/**
- * Parse the "cluster nodes" command reply to nodes dict.
- */
-static dict *parse_cluster_nodes(valkeyClusterContext *cc, valkeyReply *reply) {
-    int ret;
-    dict *nodes = NULL;
-    dict *nodes_name = NULL;
-    valkeyClusterNode *master, *slave;
-    cluster_slot *slot;
-    char *pos, *start, *end, *line_start, *line_end;
-    char *role;
-    int role_len;
-    int slot_start, slot_end, slot_ranges_found = 0;
-    sds *part = NULL, *slot_start_end = NULL;
-    int count_part = 0, count_slot_start_end = 0;
-    int k;
-    int len;
+/* Store a parsed replica node in given dict using the primary_id as key.
+ * Additional replicas for a primary are stored within the first replica. */
+static int store_replica_node(dict *replicas, char *primary_id, valkeyClusterNode *node) {
```

Comment:
Is it written anywhere that the `replicas` dict maps primary-id to a replica node, where the other replicas are added as slaves to it..? (which is a very weird way to store it; see other comment)

Reply:
I first kept the logic how to hold replica nodes while parsing, but it was weird, I have now changed it.
I have now added a new dictType to hold replica nodes while parsing.

The dict now uses primary_id as key, which points to a list of replica nodes.
Since the order in `cluster nodes` is unknown; when all nodes have been parsed the list of replicas for a given primary is moved to `primary->slaves`. (to be renamed)